### PR TITLE
feat: use `captureBeyondViewport` in `Page.captureScreenshot`

### PR DESCRIPTION
**Pull request by sadym-chromium**
BREAKING CHANGE:
- `page.screenshot` makes a screenshot with the clip dimensions, not cutting it by the ViewPort size.

### CODE REVIEW

**Comment by mathiasbynens** [comment]
> `page.screenshot` makes a screenshot with the clip dimensions, not cutting it by the ViewPort size.

Are you saying this restores the pre-v6.0.0 behavior?

File: test/screenshot.spec.ts [hunk 1]
```diff
@@ -50,26 +50,8 @@ describe('Screenshots', function () {
...
-          height: 100,
-        },
-      });
-      expect(screenshot).toBeGolden('screenshot-offscreen-clip.png');
-    });
-    // TODO: remove after the screenshot is fixed.
-    // https://crbug.com/1173457
     itFailsFirefox(
-      'should clip elements to the viewport size with artefacts',
+      'should get screenshot bigger then the viewport',
```
Comment (mathiasbynens) [inline test/screenshot.spec.ts]:
```suggestion
      'should get screenshot bigger than the viewport',
```

File: test/screenshot.spec.ts [hunk 2]
```diff
@@ -50,26 +50,8 @@ describe('Screenshots', function () {
       });
       expect(screenshot).toBeGolden('screenshot-clip-rect.png');
     });
-    // TODO: enable after the screenshot is fixed.
```
Comment (mathiasbynens) [inline test/screenshot.spec.ts:53]: Why is this test being removed?

**Comment by sadym-chromium** [comment]
@mathiasbynens 
> Are you saying this restores the pre-v6.0.0 behavior?

* Before 90, the screenshot is cut based on the ViewPort position and size. 
* In 90 without `captureBeyondViewport`, the screenshot is cut based on the ViewPort size and clip position with some unwilling artefacts out of the ViewPort.
* In 90 with `captureBeyondViewport` the screenshot is cut based on the clip position and size.

From this point, using `captureBeyondViewport` is still a breaking change, but it makes the screenshots work properly.

File: test/screenshot.spec.ts (hunk 2 above)
Comment (sadym-chromium) [reply test/screenshot.spec.ts:53]: The bug https://crbug.com/1173457 was closed with the motivation: if you need a screenshot bigger than the ViewPort, you have to use the flag `captureBeyondViewport `.

This change uses the flag, screenshot works in the new way, but in the way can easily be called "correctly", the same way Firefox does it. No further `TODO` needed.

It used to be 2 similar tests with the different expectations:
* skipped `should clip elements to the viewport size without artefacts`
* `should clip elements to the viewport size with artefacts`

I removed one of them and adjusted the expectations. It covers the issue point and the breaking change, verifying the screenshot works correctly.

**Review by mathiasbynens — APPROVED** [review]

File: test/screenshot.spec.ts (hunk 2 above)
Comment (mathiasbynens) [reply test/screenshot.spec.ts:53]: Ack, thanks!

**Comment by mahnunchik** [comment]
Why `captureBeyondViewport` option doesn't documented? https://github.com/puppeteer/puppeteer/blob/v8.0.0/docs/api.md#pagescreenshotoptions

**Comment by mahnunchik** [comment]
@sadym-chromium could you please make `captureBeyondViewport` configurable. To be able to not use this patch https://github.com/puppeteer/puppeteer/issues/7043#issuecomment-812068853

**Comment by sadym-chromium** [comment]
PR #7063 should address the issue.

**Review by jschfflr — COMMENTED** [review]
LGTM!

**Pull request merged**